resources/sts/4.9/openshift_cloud_ingress_operator_cloud_credentials_policy.json

```diff
@@ -0,0 +1,48 @@
+{
+  "Version": "2012-10-17",
+  "Statement": [
+      {
+          "Effect": "Allow",
+          "Action": [
+              "elasticloadbalancing:AddTags",
+              "elasticloadbalancing:CreateListener",
+              "elasticloadbalancing:CreateLoadBalancer",
+              "elasticloadbalancing:DeleteLoadBalancer",
+              "elasticloadbalancing:DescribeLoadBalancers",
+              "elasticloadbalancing:DescribeTags",
+              "elasticloadbalancing:DescribeTargetGroups",
+              "ec2:DescribeAccountAttributes",
+              "ec2:DescribeAddresses",
+              "ec2:DescribeInternetGateways",
+              "ec2:DescribeSecurityGroups",
+              "ec2:DescribeSubnets",
+              "ec2:DescribeVpcs",
+              "ec2:DescribeVpcClassicLink",
+              "ec2:DescribeInstances",
+              "ec2:DescribeNetworkInterfaces",
+              "ec2:DescribeClassicLinkInstances",
+              "ec2:DescribeRouteTables",
+              "ec2:AuthorizeSecurityGroupEgress",
+              "ec2:AuthorizeSecurityGroupIngress",
+              "ec2:CreateSecurityGroup",
+              "ec2:DeleteSecurityGroup",
```

Comment:
@mrWinston are these actually needed by CIO?

Reply:
Looking only at the code itself would give the following permissions:
```
{
  "Version": "2012-10-17",
  "Statement": [
      {
          "Effect": "Allow",
          "Action": [
              "elasticloadbalancing:AddTags",
              "elasticloadbalancing:CreateListener",
              "elasticloadbalancing:CreateLoadBalancer",
              "elasticloadbalancing:DeleteLoadBalancer",
              "elasticloadbalancing:DescribeLoadBalancers",
              "elasticloadbalancing:DescribeTags",
              "elasticloadbalancing:DescribeTargetGroups",
              "ec2:DescribeSubnets",
              "ec2:DescribeInstances",
              "ec2:DescribeRouteTables",
              "route53:ChangeResourceRecordSets",
              "route53:ListHostedZones",
              "route53:ListHostedZonesByName",
              "route53:ListResourceRecordSets",
          ],
          "Resource": "*"
      }
  ]
}
```

Which is vastly different to what the currently deployed CIO is allowed to do.
Either, the CIO has a lot more permissions than what's actually needed *or* i'm doing something wrong when determining the permissions.

Anyway, i'm currently checking the permissions against the non-sts version.

Reply:
The CIO does actually work with only the permissions mentioned in the previous comment. changing the publishingstrategy from internal to external and back works fine with that limited set.

Still, I'm not entirely confident that i can accurately judge the blast radius of this change.

Reply:
In non-STS mode, the cluster has Administrator Access in the AWS account, so each operator can technically get whatever permissions it wants. That doesn't mean it needs all those permissions.

Reply:
Correct. My point is the Credentials Request currently used by CIO doesn't necessarily reflect the actual minimal permissions because that hasn't been a requirement before.

Reply:
So, can i resolve this thread then? I've checked if the CIO works with the given permissions by modifying the credentials request. Is there anything else I should be doing?